32blit-sdl/Input.cpp

```diff
@@ -149,3 +196,21 @@ void Input::_virtual_analog(int x, int y, int half_w, int half_h) {
 	target->set_joystick(0, jx);
 	target->set_joystick(1, jy);
 }
+
+void Input::_add_controller(SDL_GameController* gc) {
+  // welcome rumble to test if it can rumble
+  auto can_rumble = SDL_GameControllerRumble(gc, 0xFFFF, 0xFFFF, 200);
```

Comment:
If you don't actually want the welcome rumble, you can probe for rumble functionality using short low-intensity rumble like `SDL_GameControllerRumble(gc, 1, 1, 1)` for versions of SDL that lack `SDL_GameControllerHasRumble()`. It's so short that the rumble motors won't even have time to spin up.